core/divelist.c

```diff
@@ -1757,6 +1758,20 @@ bool trip_less_than(const struct dive_trip *a, const struct dive_trip *b)
 	return comp_trips(a, b) < 0;
 }
 
+static int comp_dive_or_trip(struct dive_or_trip a, struct dive_or_trip b)
+{
+	if (a.dive && b.dive)
+		return comp_dives(a.dive, b.dive);
+	if (a.trip && b.trip)
+		return comp_trips(a.trip, b.trip);
+	return a.dive ? 1 : -1;
+}
```

Comment:
The first two cases make sense. But then the last one might deserve some explanation. So if you compare a dive to a trip (why would you ever?) the result depends on whether a is a dive? I don't get it.

Reply:
Yeah, code is bogus (as in missing something) - will fix and comment.

Reply:
Yay, I actually caught an issue :-)